src/components/sourceCards/SourceCards.tsx

```diff
@@ -41,59 +35,81 @@ export default function SourceCards({
       break;
     }
   }
+
   for (let i = 0; i < items[items.length - 1].input_slot; i++) {
-    // console.log(`On input slot: ${i + 1}`);
-    // console.log(`Checking sources:`);
-    // console.log(tempItems);
     tempItems.every((source) => {
+      const id = source._id ? source._id : '';
+      const isSource = isISource(source);
       if (source.input_slot === i + 1) {
-        // console.log(`Found source on input slot: ${i + 1}`);
-        // console.log(`Removing source "${source.name}" from sources list`);
         tempItems = tempItems.filter((i) => i._id !== source._id);
-        // console.log(`Adding source "${source.name}" to grid`);
         if (!productionSetup.isActive) {
           gridItems.push(
             <DragItem
-              key={`${source.ingest_source_name}-${source.input_slot}-key`}
-              id={source._id}
+              key={id === typeof String ? id : id.toString()}
+              id={id}
               onMoveItem={moveItem}
               previousOrder={productionSetup.sources}
-              currentOrder={currentOrder}
+              currentOrder={items as SourceReference[]}
               productionSetup={productionSetup}
               updateProduction={updateProduction}
               selectingText={selectingText}
             >
-              <SourceCard
-                source={source}
-                label={source.label}
-                src={source.src}
-                onSourceUpdate={onSourceUpdate}
-                onSourceRemoval={onSourceRemoval}
-                onSelectingText={(isSelecting: boolean) =>
-                  setSelectingText(isSelecting)
-                }
-              />
+              {isSource ? (
+                <SourceCard
+                  source={source}
+                  label={source.label}
+                  src={source.src}
+                  onSourceUpdate={onSourceUpdate}
+                  onSourceRemoval={onSourceRemoval}
+                  onSelectingText={(isSelecting) =>
+                    setSelectingText(isSelecting)
+                  }
+                  type={'ingest_source'}
+                />
+              ) : (
+                <SourceCard
+                  sourceRef={source}
+                  label={source.label}
+                  onSourceUpdate={onSourceUpdate}
+                  onSourceRemoval={onSourceRemoval}
+                  onSelectingText={(isSelecting) =>
+                    setSelectingText(isSelecting)
+                  }
+                  type={source.type}
+                />
+              )}
```

Comment:
Minor detail but instead of copying the code you can check if it is a source when passing the prop.
`src={isSource ? source.src : undefined}`. Same thing with type